### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by Adelhaide22**
Add scroll saving, refactor business logic and display manager for winforms UI

### CODE REVIEW

File: Core.Tests/StateManagerTests.cs [hunk 1]
```diff
@@ -11,42 +11,42 @@ public class StateManagerTests
...
-        [Order(2)]
-        public void Save()
-        {
-            var t1 = State.Instance.SaveTransactionsToJson();
-            var oldState = State.Instance;
-            StateManager.LoadTransactions(Enumerable.Empty<(string, Stream)>(), t1);
-
-            State.Instance.Transactions.Should().BeEquivalentTo(oldState.Transactions);
-        }
+        //[Test]
```
Comment (siriak) [inline Core.Tests/StateManagerTests.cs]: Uncomment and update tests

File: Core/Categories/CategoriesOrderer.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        {
+            var typeX = x.GetType();
+            var typeY = y.GetType();
+
+            if (typeX == typeY)
+            {
+                return x.Name.CompareTo(y.Name);
+            }
+
+            var typeToInt = new Dictionary<Type, int>
```
Comment (siriak) [inline Core/Categories/CategoriesOrderer.cs]: Make static readonly

File: Core/Categories/CategoriesOrderer.cs [hunk 3]
```diff
@@ -0,0 +1,28 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace Core.Categories
+{
+    public class CategoriesOrderer : IComparer<Category>
+    {
+        public int Compare(Category x, Category y)
```
Comment (siriak) [inline Core/Categories/CategoriesOrderer.cs]: Add tests

File: Core/IRepository.cs [hunk 4]
```diff
@@ -0,0 +1,27 @@
+﻿using Core.Categories;
+using System.Collections.Generic;
+using System.IO;
+
+namespace Core
+{
+    public interface IRepository
+    {
+        public void SaveAutoCategoriesToFile();
```
Comment (siriak) [inline Core/IRepository.cs]: Why public?

Comment (siriak) [reply Core/IRepository.cs]: Why no parameters?

Comment (siriak) [reply Core/IRepository.cs]: Why ToFile in name?

File: WinFormsUI/DisplayManager.cs [hunk 5]
```diff
@@ -10,5 +12,24 @@ public static string FormatLedgerRecord(Transaction transaction, IEnumerable<str
             return $"{transaction.Date.ToString("yyyy.MM.dd")}:" + $"{transaction.Amount.Amount}".PadLeft(10) 
                 + $" {transaction.Amount.Currency} [{string.Join(", ", categories)}] {transaction.Description}";
         }
+
+        public static string AddPrefixToCategory(double todayRelative, Category c)
+        {
+            var level = todayRelative switch
```
Comment (siriak) [inline WinFormsUI/DisplayManager.cs]: Move to BLL

File: WinFormsUI/DisplayManager.cs [hunk 6]
```diff
@@ -10,5 +12,24 @@ public static string FormatLedgerRecord(Transaction transaction, IEnumerable<str
             return $"{transaction.Date.ToString("yyyy.MM.dd")}:" + $"{transaction.Amount.Amount}".PadLeft(10) 
                 + $" {transaction.Amount.Currency} [{string.Join(", ", categories)}] {transaction.Description}";
         }
+
+        public static string AddPrefixToCategory(double todayRelative, Category c)
```
Comment (siriak) [inline WinFormsUI/DisplayManager.cs]: GetPrefix(Level l)

File: WinFormsUI/DisplayManager.cs [hunk 7]
```diff
@@ -10,5 +12,24 @@ public static string FormatLedgerRecord(Transaction transaction, IEnumerable<str
...
+            {
+                _ when todayRelative <= 0 => Levels.Empty,
+                _ when todayRelative <= 0.1 => Levels.Low,
+                _ when todayRelative < 1 => Levels.Normal,
+                _ => Levels.Full,
+            };
+
+            return level switch
+            {
+                _ when Equals(level, Levels.Empty) => "(EMPTY) ",
```
Comment (siriak) [inline WinFormsUI/DisplayManager.cs]:
```suggestion
                Levels.Empty => "(EMPTY) ",
```

File: WinFormsUI/TransactionEditor.Designer.cs [hunk 8]
```diff
@@ -100,7 +103,7 @@ private void InitializeComponent()
             this.Controls.Add(this.txtboxDescription);
             this.Controls.Add(this.btnSave);
         }
-
+
```
Comment (siriak) [inline WinFormsUI/TransactionEditor.Designer.cs]: Spaces

File: WinFormsUI/TransactionEditor.cs [hunk 9]
```diff
@@ -14,6 +14,14 @@ public TransactionEditor(Transaction transaction)
             InitializeComponent();
         }
 
+        private void CheckEnterKeyPress(object sender, KeyPressEventArgs e)
+        {
+            if (e.KeyChar == (char)Keys.Enter)
+            {
+                btnSave_Clicked(null, null);
```
Comment (siriak) [inline WinFormsUI/TransactionEditor.cs]: Move to Save method and call it from btnSave_Clicked and here

**Review by siriak — CHANGES_REQUESTED** [review]

File: Core/IRepository.cs (hunk 4 above)
Comment (Adelhaide22) [reply Core/IRepository.cs]: public - because it's used in another assembly (WinFormsUI)

**Review by siriak — APPROVED** [review]

**Pull request merged**